libtenzir/builtins/aggregation-functions/all.cpp

```diff
@@ -51,7 +56,40 @@ class all_function final : public aggregation_function {
   std::optional<bool> all_ = {};
 };
 
-class plugin : public virtual aggregation_function_plugin {
+class all_instance final : public aggregation_instance {
+public:
+  explicit all_instance(ast::expression expr) : expr_{std::move(expr)} {
+  }
+
+  auto update(const table_slice& input, session ctx) -> void override {
+    auto arg = eval(expr_, input, ctx);
+    if (all_ and all_.value() == false) {
+      return;
+    }
```

Comment:
I think it'd be better if we still do the type check even if we already know the result.